superset-frontend/packages/superset-ui-chart-controls/src/operators/pivotOperator.ts

```diff
@@ -31,11 +31,13 @@ export const pivotOperator: PostProcessingFactory<PostProcessingPivot> = (
 ) => {
   const metricLabels = ensureIsArray(queryObject.metrics).map(getMetricLabel);
   const { x_axis: xAxis } = formData;
+
   if ((xAxis || queryObject.is_timeseries) && metricLabels.length) {
+    const index = [getColumnLabel(formData.x_axis || DTTM_ALIAS)];
```

Comment:
```suggestion
    const index = [getColumnLabel(xAxis || DTTM_ALIAS)];
```

Reply:
This or we can remove extracting xAxis from formData.x_axis in line 33

Reply:
Oops, good catch! 👍